### PR TITLE
horses race mvp done

### LINKED ISSUE #12: Horses race game
# REQUIREMENTS
A horses race online game with chat

There will be 5 horses
Players could choose 1 horse and bet on that horse

### PR DESCRIPTION
**Pull request by alanRazoMen**
# Description
We carry out the horse race minigame, the players choose the horse number and the amount they wish to bet, then the race begins and if the horse you chose wins, you will be paid x3 of the bet

 
 
image

image

### CODE REVIEW

File: app/channels/classes/HorsesGame.rb [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    emit({type: "updateBets", bets: get_bets})
+  end
+
+  def reset
+    @phase = "betsTime"
+    @horses.each {|h| h.reset}
+    @bets = []
+    emit({type: "reset"})
+  end
+end
```
Comment (Frcomparan) [inline app/channels/classes/HorsesGame.rb]: Add a blank line here

File: app/channels/classes/Horse.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  end
+
+  def win?
+    @pos >= 1000
+  end
+
+  def reset
+    @pos = 0
+  end
+end
```
Comment (Frcomparan) [inline app/channels/classes/Horse.rb]: Add a blank line here

File: app/assets/stylesheets/slot_machine.css [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  background-color: white;
+}
+
+.barra div {
+  box-sizing: border-box;
+  height: 200px;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  padding: 10px;
```
Comment (Frcomparan) [inline app/assets/stylesheets/slot_machine.css]: Sort this styles

File: app/assets/stylesheets/slot_machine.css [hunk 4]
```diff
@@ -0,0 +1,36 @@
+.slot_machine{
+  border: 5px outset red;
+  height: 200px;
+  overflow: hidden;
+  display: grid;
+  gap: 5px;
+  grid-template-columns: repeat(3, 1fr);
```
Comment (Frcomparan) [inline app/assets/stylesheets/slot_machine.css]: Sort this styles

File: app/assets/stylesheets/horses_race.css [hunk 5]
```diff
@@ -0,0 +1,89 @@
+.horses-race {
+  width: 100%;
+  height: 260px;
+  display: flex;
```
Comment (Frcomparan) [inline app/assets/stylesheets/horses_race.css]: Sort this styles, and the rest of this file

File: app/assets/stylesheets/horses_race.css [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+}
+
+.start{
+  min-width: 100px;
+  height: 100%;
+  background-color: orange;
```
Comment (Frcomparan) [inline app/assets/stylesheets/horses_race.css]: Here

File: app/assets/stylesheets/horses_race.css [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+}
+
+.track{
+  position: relative;
+  border-left: 5px dashed;
+  border-right: 5px dashed;
+  background-color: green;
+  width: 100%;
+  height: 100%;
```
Comment (Frcomparan) [inline app/assets/stylesheets/horses_race.css]: Here

File: app/assets/stylesheets/horses_race.css [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+canvas{
+  width: 100%;
+  background-color: green;
+}
+
+.horse {
+  width: 100px;
+  left: calc(0% - 100px);
+  position: absolute;
+  transition: 2s linear;
```
Comment (Frcomparan) [inline app/assets/stylesheets/horses_race.css]: Here

File: app/assets/stylesheets/horses_race.css [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+  gap: 3px;
+  font-size: 25px;
+}
+
+.bet #horseSelection div{
+  border: 1px solid white;
+  width: 50px;
+  text-align: center;
+  border-radius: 10px;
+  cursor: pointer;
```
Comment (Frcomparan) [inline app/assets/stylesheets/horses_race.css]: Here

**Review by Frcomparan — CHANGES_REQUESTED** [review]

**Review by Frcomparan — APPROVED** [review]
LGTM

**Review by brocha05 — APPROVED** [review]
LGTM

**Pull request merged**